### ISSUE TITLE
Katago makes suicide move?

### ISSUE THREAD

**Issue by hwj-111** [lightvector/KataGo#554]
I set my katago with taylor-tromp rule and played on cgos, it was fine for quite long time until recently.

There are some CrazyStone bots with a play style to put stones until there is no way to continue ... during that period, katago put suicide move in couple of games like below and lost the games due to illegal moves

http://www.yss-aya.com/cgos/viewer.cgi?19x19/SGF/2021/09/27/779150.sgf
http://www.yss-aya.com/cgos/viewer.cgi?19x19/SGF/2021/09/24/778806.sgf 

Could this be a bug?

**Comment by lightvector** [comment]
Suicide is legal in tromp-taylor rules, so no this is not a bug. You should set the rules to be whatever the correct rules are.

**Comment by hwj-111** [comment]
Oh, it is a cgos bug then

**Comment by lightvector** [comment]
Hehe, it's probably not a cgos bug. I bet they don't use tromp-taylor rules.

**Comment by lightvector** [comment]
Yep, here: http://www.yss-aya.com/cgos/ see CGOS Rules of Play. The server explicitly states that it "very similar to the Tromp/Taylor rules (sometimes referred to as "The Logical Rules") **but prohibit suicide**" (bold mine). So just configure the individual rules options KataGo in the config to match and you should be good to go.

**Comment by hwj-111** [comment]
I see ... my fault

**Comment by lightvector** [comment]
Eh, pretty minor. Self-capture definitely doesn't come up very much in normal games. I guess KataGo could try to clean up more efficiently too. :)

**Comment by michito744** [comment]
It seems that the suicide move will be effective only in the situation where the ko occurs.

example:-)
![2021-09-27 (2)](https://user-images.githubusercontent.com/53891655/134929591-08b4dd41-e181-403b-873a-4ba4f0690dd4.png)
